src/Effect/Monad/Writer/Transformer/Base.agda

```diff
@@ -0,0 +1,44 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Basic type and definition of the writer monad transformer
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Effect.Monad.Writer.Transformer.Base {w} (W : Set w) where
+
+open import Data.Product using (_×_; _,_)
+open import Data.Unit.Polymorphic using (⊤; tt)
+open import Function.Base using (id)
+open import Level using (Level; suc; _⊔_)
+
+private
+  variable
+    g : Level
+    A : Set w
+
+------------------------------------------------------------------------
+-- Writer monad operations
+
+record RawMonadWriter
+       (M : Set w → Set g)
+       : Set (suc w ⊔ g) where
+  field
+    writer : W × A → M A
+    listen : M A → M (W × A)
+    pass   : M ((W → W) × A) → M A
+
+  tell : W → M ⊤
+  tell w = writer (w , tt)
+
+------------------------------------------------------------------------
+-- Writer monad transformer
+
+record WriterT
+       (M : Set w → Set g)
+       (A : Set w)
+       : Set (w ⊔ g) where
+  constructor mkWriterT
+  field runWriterT : M (W × A)
```

Comment:
Can we please also get
```
evalWriterT : RawFunctor M → WriterT W M A → M A
execWriterT : RawFunctor M → WriterT W M A → M W
```
(and the equivalent variant without `T` in `Effect.Monad.Writer`?